pkg/logs/input/kubernetes/scanner.go

```diff
@@ -136,12 +138,13 @@ func (s *Scanner) getSource(pod *kubelet.Pod, container kubelet.ContainerStatus)
 		}
 	} else {
 		cfg = &config.LogsConfig{
+			Type: config.FileType,
+			Path: s.getPath(pod, container)
 			Source:  kubernetesIntegration,
 			Service: kubernetesIntegration,
 		}
 	}
-	cfg.Type = config.FileType
-	cfg.Path = s.getPath(pod, container)
```

Comment:
why are we moving these two lines?

Reply:
Because we need the type to be set for the validation to succeed